### PR TITLE
ITE: riscv: it8xxx2: declare gctrl dts node and registers

### PR DESCRIPTION
**Pull request by GTLin08**
Add general control(gctrl) node in dts, and declare gctrl registers

---
This change is [Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/37762)

### CODE REVIEW

**Review by brockus-zephyr — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/37762)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jackrosenthal, @katsuster, @keith-zephyr, @kgugala, @mgielda, @pgielda, and @sjg20)

File: soc/riscv/riscv-ite/common/chip_chipregs.h [hunk 1]
```diff
@@ -1940,4 +1940,79 @@ enum chip_pll_mode {
...
+ *
+ * (20xxh) General Control (GCTRL) registers
+ *
+ */
+#ifndef __ASSEMBLER__
+struct gctrl_it8xxx2_regs {
+	/* 0x00-0x01: Reserved1 */
+	volatile uint8_t reserved1[2];
+	/* 0x02: Chip Version */
+	volatile uint8_t GCTRL_ECHIPVER;
```
Comment (keith-zephyr) [inline soc/riscv/riscv-ite/common/chip_chipregs.h:1953]: Please add some build asserts to validate the offsets are the expected value.  This ensures any future changes to the register map don't break the existing register locations.

Example here:
https://github.com/zephyrproject-rtos/zephyr/blob/5d988fde057f7d69d4e0ee658407326706f0f31b/soc/arm/nuvoton_npcx/common/reg/reg_def.h#L20
https://github.com/zephyrproject-rtos/zephyr/blob/5d988fde057f7d69d4e0ee658407326706f0f31b/soc/arm/nuvoton_npcx/common/registers.c#L12

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-ite/common/chip_chipregs.h (hunk 1 above)
Comment (GTLin08) [reply soc/riscv/riscv-ite/common/chip_chipregs.h:1969]: Done

**Review by keith-zephyr — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/37762)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jackrosenthal, @katsuster, @kgugala, @mgielda, @pgielda, and @sjg20)

**Pull request merged**